pkg/services/rendering/plugin_mode.go

```diff
@@ -22,7 +23,7 @@ func (rs *RenderingService) renderViaPlugin(ctx context.Context, opts Opts) (*Re
 		return nil, err
 	}
 
-	ctx, cancel := context.WithTimeout(ctx, opts.Timeout)
+	ctx, cancel := context.WithTimeout(ctx, opts.Timeout+time.Second*2)
```

Comment:
Would it make sense to explain the reason for doubling the timeout with a comment here?

Reply:
I noticed [http mode](https://github.com/grafana/grafana/blob/master/pkg/services/rendering/http_mode.go#L63) and [phantomjs](https://github.com/grafana/grafana/blob/master/pkg/services/rendering/phantomjs.go#L66) used that, but not the plugin mode. I guess the reasoning is that the image renderer service, plugin and phantomjs receives the timeout and times out after that time. By adding 2 additional seconds Grafana gives the image renderer service, plugin and phantomjs to timeout and return possible error message.

Reply:
@marefr Sorry, I meant a comment in the code, as a hint to future readers :D

Reply:
Ugh, I realize now I even misread the code, I thought that was a multiplication by 2 :man_facepalming:

Reply:
I added comments